### PR TITLE
Fix etl

### PR DESCRIPTION
**Pull request by alrocar**
I've re-tested the etl script. Here the list of changes:

- Added support for files with a different encoding than `utf-8`
- Added support for uploading files with no geometry
- Fixed some bugs
- Added a couple of sample files to test
- Extracted some code to common methods since there was duplicated code blocks

Adding @danicarrion for CR

cc/ @AbelVM @ramiroaznar

### CODE REVIEW

File: etl.py [hunk 1]
```diff
@@ -49,55 +56,115 @@ def chunks(full_list, chunk_size, start_chunk=1, end_chunk=None):
...
+        return longitude
+
+    def get_latitude(self, record):
+        latitude = self.get_coord(record, self.y_column)
+        if abs(latitude) > MAX_LAT:
+            return None
+        return latitude
+
+    def get_coord(self, record, type):
+        coord = record[type] or DEFAULT_COORD
```
Comment (danicarrion) [inline etl.py]: possible KeyError

File: etl.py [hunk 2]
```diff
@@ -49,55 +56,115 @@ def chunks(full_list, chunk_size, start_chunk=1, end_chunk=None):
...
+        try:
+            coord = float(coord)
+        except ValueError:
+            coord = DEFAULT_COORD
```
Comment (danicarrion) [inline etl.py]: not really that important, but for the sake of consistency, I think it should read `coord = float(DEFAULT_COORD)`

File: etl.py [hunk 3]
```diff
@@ -49,55 +56,115 @@ def chunks(full_list, chunk_size, start_chunk=1, end_chunk=None):
...
 
+    def create_geom_query(self, record):
+        null_result = NULL_VALUE + ","
+        if self.force_no_geometry:
+            return null_result
+
+        longitude = self.get_longitude(record)
+        latitude = self.get_latitude(record)
+
+        if longitude is None or latitude is None:
```
Comment (danicarrion) [inline etl.py]: what if `longitude` or `latitude` are `DEFAULT_COORD`? is that also something to consider?

File: etl.py [hunk 4]
```diff
@@ -18,12 +18,19 @@
...
+FILE_ENCODING = config.get('etl', 'file_encoding')
 CHUNK_SIZE = int(config.get('etl', 'chunk_size'))
 MAX_ATTEMPTS = int(config.get('etl', 'max_attempts'))
+FORCE_NO_GEOMETRY = config.getboolean('etl', 'force_no_geometry')
 
 api_auth = APIKeyAuthClient(CARTO_BASE_URL, CARTO_API_KEY)
 sql = SQLClient(api_auth)
 bsql = BatchSQLClient(api_auth)
+UTF8 = "utf-8"
+DEFAULT_COORD = 0
```
Comment (danicarrion) [inline etl.py]: aren't you using this value for incorrect geometries too? wouldn't that conflict with a legit 0 for lat or lon?

File: etl.py [hunk 5]
```diff
@@ -49,55 +56,115 @@ def chunks(full_list, chunk_size, start_chunk=1, end_chunk=None):
...
+        try:
+            value = record[column]
+            if value is None or not value.strip():
+                result = NULL_VALUE + ","
+            else:
+                float(record[column])
+        except ValueError:
+            result = "'{value}',".format(value=value)
+        else:
+            if result is None:
```
Comment (danicarrion) [inline etl.py]: Does this code ever get executed?

File: etl.py [hunk 6]
```diff
@@ -49,55 +56,115 @@ def chunks(full_list, chunk_size, start_chunk=1, end_chunk=None):
...
+                result = "{value},".format(value=value)
+        return result
```
Comment (danicarrion) [inline etl.py]: Wouldn't this be a cleaner way to achieve the same result?

```python
value = record[column]  # You might want to check for KeyError here too

try:
    return "{value},".format(value=float(value))
except (ValueError, TypeError):
    if value is None or not value.strip():
        value = NULL_VALUE
    return "'{value}',".format(value=value)
```

File: etl.py (hunk 4 above)
Comment (AbelVM) [reply etl.py]: `DEFAULT_COORD` should be `None`, IMHO

Comment (ramiroaznar) [reply etl.py]: +1

Comment (alrocar) [reply etl.py]: 👍

File: etl.py (hunk 3 above)
Comment (alrocar) [reply etl.py]: yep, adding the condition as well

File: etl.py (hunk 6 above)
Comment (alrocar) [reply etl.py]: 👍 Modified a bit since we are missing the case when we want return just NULL (no the 'NULL' string)

File: etl.py (hunk 1 above)
Comment (alrocar) [reply etl.py]: 👍

File: etl.py (hunk 2 above)
Comment (alrocar) [reply etl.py]: setting DEFAULT_COORD as None as per a previous comment

**Comment by alrocar** [comment]
I've added some fixes after your review @danicarrion 

I think in the next cycle we should add tests to check that the queries generated are well-formed and so on.

Ready to merge when you approve :)

**Pull request merged**